src/org/rust/lang/core/grammar/rust.bnf

```diff
@@ -446,10 +446,10 @@ enum_args ::= LBRACE struct_decl_fields COMMA? RBRACE
 // Trait
 ///////////////////////////////////////////////////////////////////////////////////////////////////
 
-trait_item ::= UNSAFE? TRAIT IDENTIFIER generic_params for_sized? type_param_bounds? where_clause? LBRACE trait_contents* RBRACE
+trait_item ::= UNSAFE? TRAIT IDENTIFIER generic_params? for_sized? type_param_bounds? where_clause? LBRACE trait_contents* RBRACE
```

Comment:
this way I can just check `if (genericParams != null)` and don't have to check its child nodes